### ISSUE TITLE
Rule proposal: `prefer-native-coercion-functions`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1643]
### Description

I want to propose a more general rule to replace #1564.

[Primitive wrapper objects in JavaScript](https://developer.mozilla.org/en-US/docs/Glossary/Primitive)

When function is equivalence to the primitive wrapper, it should be used directly, including `String`, `Number`, `BigInt`, `Boolean`, and `Symbol `.

### Fail

```js
const toString = (v) => String(v);
```

```js
const toString = (v) => `${v}`;
```

```js
const toString = (v) => '' + v;
```

```js
const toNumber = (v) => Number(v);
```

```js
const toNumber = (v) => +v;
```

```js
const toBoolean = (v) => Boolean(v);
```

```js
const toBoolean = (v) => !!v;
```

```js
// These functions don't have to be arrow function or function expression/declaration
useFunction(function (v) {
	return Boolean(v);
});
```

```js
/* 
These callbacks not really equivalence to `Boolean`, but the result is used as `Boolean`,
Addtional check to cover cases in #1564
*/
list.every(v => v);
list.filter(v => v);
list.some(v => v);
list.find(v => v);
list.findIndex(v => v);
```

### Pass

```js
const toString = String;
```

```js
useFunction(Boolean);
```

```js
list.every(Boolean);
list.filter(Boolean);
list.some(Boolean);
list.find(Boolean);
list.findIndex(Boolean);
```

Some cases already coverd by builtin rule [no-implicit-coercion](https://eslint.org/docs/rules/no-implicit-coercion), so maybe we can assume some type-casting is already in shape of `v => Boolean(v)`

**Comment by fisker** [comment]
@B2o5T

**Comment by B2o5T** [comment]
Really love how this was transformed from simple `.filter(Boolean)` to this 👍

**Comment by sindresorhus** [comment]
> Primitive wrapper objects in JavaScript

Not exactly. `new String(value)` is a primitive wrapper object. `String(value)`, however, is just a coercion. So the `prefer-primitive-wrapper` name would not be correct. I don't think `prefer-primitive-function` is an optimal name either.

How about `prefer-native-coercion-functions`?

**Comment by fisker** [comment]
Accepted.
